cmd/thanos/query.go

```diff
@@ -39,7 +40,7 @@ import (
 func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string) {
 	cmd := app.Command(name, "query node exposing PromQL enabled Query API with data retrieved from multiple store nodes")
 
-	grpcBindAddr, httpBindAddr, srvCert, srvKey, srvClientCA, newPeerFn := regCommonServerFlags(cmd)
+	grpcBindAddr, httpBindAddr, webRoutePrefix, webExternalPrefix, webPrefixHeaderName, srvCert, srvKey, srvClientCA, newPeerFn := regCommonServerFlags(cmd)
```

Comment:
Please don't set these as common flags,  these are useless for other components exclude query and rule. You should use these only in query and rule.